OpenRA.Game/Network/UnitOrders.cs

```diff
@@ -65,9 +65,40 @@ public static void ProcessOrder(OrderManager orderManager, World world, int clie
 						var client = orderManager.LobbyInfo.ClientWithIndex(clientId);
 						if (client != null)
 							client.State = Session.ClientState.Disconnected;
+
 						break;
 					}
 
+				case "NewAdmin": /* Find a new admin and hand bots over mid-game. */
+					{
+						var client = orderManager.LobbyInfo.ClientWithIndex(clientId);
+						if (client == null)
+							break;
+
+						if (orderManager.LobbyInfo.GlobalSettings.Dedicated && client.IsAdmin)
+						{
+							var nextAdmin = orderManager.LobbyInfo.Clients.Where(c => c.Bot == null && c.Index != client.Index).MinByOrDefault(c => c.Index);
+							if (nextAdmin != null)
+							{
+								nextAdmin.IsAdmin = true;
+								Game.AddChatLine(Color.White, "Server", "{0} is now the admin.".F(nextAdmin.Name));
+
+								var bots = orderManager.LobbyInfo.Clients.Where(c => c.Bot != null && c.BotControllerClientIndex == client.Index);
+								foreach (var bot in bots)
+								{
+									bot.BotControllerClientIndex = nextAdmin.Index;
+									var botPlayer = orderManager.World.FindPlayerByClient(bot);
+									botPlayer.ActivateBot(bot.Bot);
+									Game.Debug("Bot {0} is now controlled by {1}.".F(bot.Name, nextAdmin.Name));
```

Comment:
I'd move this to debug.log as well. Players probably don't care about these implementation details.